sched/sched/sched_note_buffer.c

```diff
@@ -0,0 +1,471 @@
+/****************************************************************************
+ * sched/sched/sched_note_buffer.c
```

Comment:
We can merge this file into note_driver to simplify the interaction.

Reply:
If merged, the exiting sched note interface `sched_note_get()` is moved into note_driver. Is it OK?